src/basic/util.c

```diff
@@ -96,8 +96,13 @@ bool in_initrd(void) {
                 saved_in_initrd = r > 0;
         }
 
-        if (saved_in_initrd != 0)
-                saved_in_initrd = access("/etc/initrd-release", F_OK) >= 0;
+        r = access("/etc/initrd-release", F_OK);
+        if (r >= 0) {
+                if (saved_in_initrd == 0)
+                        log_debug("/etc/initrd-release exists, but it's not an initrd.");
+                else
+                        saved_in_initrd = 1;
+        }
```

Comment:
now that you keep adding debug log messages, you could even add one more:

if (access(…)  >= 0) {
        …
} if (errno != ENOENT)
        log_debug_errno(errno, "Failed to test if /etc/initrd-release exists");
```

(i.e. we only expect ENOENT as error, all other errros indicate something weird going on we probably should make discoverable)

Reply:
Thanks for the suggest! I'll fix it along with another error, I forgot to fully covered the else case. now it should be good.